src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/EventRoute.cs

```diff
@@ -321,7 +321,7 @@ public void PushBranchNode(object node, object source)
             branchNode.Node = node;
             branchNode.Source = source;
             
-            BranchNodeStack.Push(branchNode);
+            (_branchNodeStack ??= new Stack<BranchNode>()).Push(branchNode);
```

Comment:
The original Stack was created with a size of 1. I think you overlooked that?

Reply:
Thanks.